glue/sales_val_flagging.py

```diff
@@ -152,6 +149,66 @@ def group_size_adjustment(df, stat_groups: list, min_threshold, condos: bool):
         df: dataframe with newly manually adjusted outlier values.
 
     """
+
+    # Assign sv_outlier_reasons
+    df["sv_outlier_reason1"] = "Null"
+    df["sv_outlier_reason2"] = "Null"
+    df["sv_outlier_reason3"] = "Null"
+
+    cols_to_iterate = [col for col in df.columns if col.startswith("sv_ind")]
+
+    # - - -
+    # TODO: try numba speed up here
+    # - - -
+    def fill_outlier_reasons(row):
+        reasons = [
+            row[f"sv_outlier_reason{i}"] for i in range(1, 4)
+        ]  # Collect current reasons
+
+        for col in cols_to_iterate:
+            if row[col]:  # If the condition is True
+                reason = col  # Use column name as reason
+                # Check if reason is not already recorded and there's a 'Null' slot to fill
+                if reason not in reasons and "Null" in reasons:
+                    next_index = reasons.index("Null") + 1  # Find the next 'Null' index
+                    row[f"sv_outlier_reason{next_index}"] = reason
+                    reasons[next_index - 1] = reason  # Update reasons list
+
+        return row
+
+    df = df.apply(fill_outlier_reasons, axis=1)
+
+    outlier_type_crosswalk = {
+        "sv_ind_ptax_flag_w_deviation": "PTAX-203 Exclusion",
+        "sv_ind_price_high_price": "High price",
+        "sv_ind_price_low_price": "Low price",
+        "sv_ind_price_high_price_sqft": "High price per square foot",
+        "sv_ind_price_low_price_sqft": "Low price per square foot",
+        "sv_ind_char_short_term_owner": "Short-term owner",
+        "sv_ind_char_family_sale": "Family Sale",
+        "sv_ind_char_non_person_sale": "Non-person sale",
+        "sv_ind_char_statistical_anomaly": "Statistical Anomaly",
+        "sv_ind_char_price_swing_homeflip": "Price swing / Home flip",
+    }
+
+    for column in ["sv_outlier_reason1", "sv_outlier_reason2", "sv_outlier_reason3"]:
+        df[column] = df[column].replace(outlier_type_crosswalk)
+
+    # Assign outlier status
+    values_to_check = {
+        "PTAX-203 Exclusion",
+        "High price",
+        "Low price",
+        "High price per square foot",
+        "Low price per square foot",
+    }
+    df["sv_is_outlier"] = np.where(df["sv_outlier_reason1"].isin(values_to_check), 1, 0)
+
+    # - - -
+    # Handle group threshold adjustment
+    # TODO: Explain choice of sv_outlier_reason$n, do we set all reasons to 0 as well?
+    # - - -
+
```

Comment:
This is an open question I want some advice on. Currently when we are manually setting the outlier status from outiler to non-outlier based on the group threshold, we only change `sv_is_outlier` from 1 to 0. We leave all of the `sv_outlier_reason$n` columns as they are. So they could have a 'High price' and a 'Family sale' status and still be 0 for `sv_is_outlier`. I was leaning towards leaving this as it is still potentially useful information, but I could see it as potentially confusing.

Reply:
Hmm, this is a good question! I think we should default to preserving all of the fields, but maybe what we need to make things clearer is an additional reason code like "Group size too small to flag" that we can add in cases like these. I'll raise this question with Dan and we can settle on a path forward.